test/login/login-test.js

```diff
@@ -0,0 +1,19 @@
+const chai = require('chai');
+
+describe('Login tests', function() {
+  // Setting this.timeout(0) disables Mocha from timing out for all
+  // tests under that describe().
+  // Set it to another value (in milliseconds), if you know the CRM
+  // shouldn't take longer than that amount of time for any operation in this file.
+  this.timeout(0);
+
+  it('allows User see their ID in top right after login', async function() {
+    await page.click('#navTabButtonUserInfoLinkId');
+    await page.waitForSelector('#navTabButtonUserInfoUserNameId');
+    const text = await page.$eval("#navTabButtonUserInfoUserNameId", (elem) => {
+      return elem.textContent;
+    });
+    const expectedUsername = process.env.ZAP_USERNAME.replace('@planning.nyc.gov', ' #').toLowerCase();
+    chai.assert(text.toLowerCase() == expectedUsername, 'Username is not in the top right profile dropdown');
+  });
+});
```

Comment:
What we can do with the test helpers below (teardown and setup-login) is wrap them in a "beforeEach" and then import them into specific tests.

Reply:
Hmm, I had that architecture originally but was thinking most of the tests would need to done after signing into CRM anyway? Also it seemed it would speed up the tests if each one didn't need to start by signing into CRM .... is performance a concern here?

Reply:
Yep, you're right — speed is an issue here.